### DISCUSSION TITLE
trace execv full arguments list

### DISCUSSION THREAD

**Discussion by hossbeast** [bpftrace/bpftrace#2309] [General]
I have come to the surprising conclusion that what I am trying to do is not possible with bpf. Can someone confirm my understanding?

I am just using `execsnoop` to trace new processes

```
BPFTRACE_STRLEN=1024 sudo bpftrace /usr/share/bpftrace/tools/execsnoop.bt
```

But the processes I am interested in have both very large individual arguments, and very large argument lists. execsnoop seems to only show the first 16 arguments.

Is there some way I can invoke `bpftrace`, or some change I can make in `execsnoop.bt` to accommodate this use case?

**Comment by fbs** [comment]
The `join` function responsible for this was created before bpf had loop support, so it has to unroll the loop. The unroll is currently hardcoded and limited to 16 (https://github.com/iovisor/bpftrace/blob/03b46c031b25523b9c5366fac4be49f316c2a541/src/bpftrace.h#L171).

As a quick fix you can up the limit and recompile.

Longer term we should revise this, add an env var to control the limit and look into bpf loops to support this

**Comment by hossbeast** [comment]
Ok, thanks! I will try increasing this limit.
